apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Converters/EnumConverter.cs

```diff
@@ -26,94 +26,65 @@ namespace Google.Cloud.Firestore.Converters
-            {
-                TypeCode underlyingTypeCode = BclType.GetTypeCode(type.GetTypeInfo().GetEnumUnderlyingType());
-                if (!s_genericTypes.TryGetValue(underlyingTypeCode, out var genericConverterType))
-                {
-                    throw new InvalidOperationException($"Unexpected underlying type code for enum: {underlyingTypeCode}");
-                }
-                return (PrimitiveConverter) Activator.CreateInstance(genericConverterType.MakeGenericType(type));
-            }
-
-            private sealed class ByteEnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => checked((byte) value);
-                internal override object Int64ToEnum(long value) => (T) (object) checked((byte) value);
-            }
-
-            private sealed class SByteEnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => checked((sbyte) value);
-                internal override object Int64ToEnum(long value) => (T) (object) checked((sbyte) value);
-            }
-
-            private sealed class Int16EnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => checked((short) value);
-                internal override object Int64ToEnum(long value) => (T) (object) checked((short) value);
-            }
-
-            private sealed class UInt16EnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => checked((ushort) value);
-                internal override object Int64ToEnum(long value) => (T) (object) checked((ushort) value);
-            }
-
-            private sealed class Int32EnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => checked((int) value);
-                internal override object Int64ToEnum(long value) => (T) (object) checked((int) value);
-            }
-
-            private sealed class UInt32EnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => checked((uint) value);
-                internal override object Int64ToEnum(long value) => (T) (object) checked((uint) value);
-            }
+            object baseValue = Int64ToEnumBaseType(value);
+            return Enum.ToObject(TargetType, baseValue);
+        }
 
-            private sealed class Int64EnumConverter<T> : PrimitiveConverter
-            {
-                internal override long EnumToInt64(object value) => (long) value;
-                internal override object Int64ToEnum(long value) => (T) (object) value;
+        public override Value Serialize(SerializationContext context, object value) =>
+            new Value { IntegerValue = EnumToInt64(value) };
+
+        private long EnumToInt64(object value) {
```

Comment:
Please fix the formatting to "brace at start of line" here and in the next line.
For the cases, I'm fine with the existing formatting, or the return statement could be on the same line as the case.

Reply:
(And all of this for Int64ToEnumBaseType too :)

Reply:
Done.  The return statements on same line look nice.  Thanks.